### ISSUE TITLE
jdk.vm.ci.meta.ValueKind$IllegalKind cannot be cast to class com.oracle.svm.core.graal.llvm.util.LLVMUtils$LLVMKind 

### ISSUE THREAD

**Issue by jperedadnr** [oracle/graal#3788]
**Describe the issue**

Compiling a native image for iOS fails with any recent GraalVM dev build version built with JDK17 (but works fine with JDK11), with the exception:

```
[hello.helloworld:13621]    (compile):  14,225.73 ms,  4.31 GB
[hello.helloworld:13621]      compile:  16,964.60 ms,  4.31 GB
Fatal error:org.graalvm.compiler.graph.GraalGraphError: java.lang.ClassCastException: class jdk.vm.ci.meta.ValueKind$IllegalKind cannot be cast to class com.oracle.svm.core.graal.llvm.util.LLVMUtils$LLVMKind (jdk.vm.ci.meta.ValueKind$IllegalKind is in module jdk.internal.vm.ci of loader 'bootstrap'; com.oracle.svm.core.graal.llvm.util.LLVMUtils$LLVMKind is in unnamed module of loader 'app')
        at node: 27|Constant(illegal, ILLEGAL) { isDefaultStable=false, stableDimension=0, rawvalue=illegal, stampKind=ILLEGAL, stamp=ILLEGAL, value=illegal,  }
        at method: byte[] com.sun.crypto.provider.GHASH.digest()  [Direct call from int GaloisCounterMode$GCMDecrypt.doFinal(byte[], int, int, byte[], int)]
        at com.oracle.svm.core.graal.llvm.NodeLLVMBuilder.doBlock(NodeLLVMBuilder.java:291)
        at com.oracle.svm.core.graal.llvm.SubstrateLLVMBackend.generate(SubstrateLLVMBackend.java:148)
        at com.oracle.svm.core.graal.llvm.SubstrateLLVMBackend.emitLLVM(SubstrateLLVMBackend.java:124)
        at com.oracle.svm.core.graal.llvm.SubstrateLLVMBackend.emitBackEnd(SubstrateLLVMBackend.java:104)
        at jdk.internal.vm.compiler/org.graalvm.compiler.core.GraalCompiler.compile(GraalCompiler.java:146)
        at jdk.internal.vm.compiler/org.graalvm.compiler.core.GraalCompiler.compileGraph(GraalCompiler.java:130)
        at com.oracle.svm.hosted.code.CompileQueue.defaultCompileFunction(CompileQueue.java:1210)
        at com.oracle.svm.hosted.code.CompileQueue.doCompile(CompileQueue.java:1172)
        at com.oracle.svm.hosted.code.CompileQueue$CompileTask.run(CompileQueue.java:279)
        at com.oracle.graal.pointsto.util.CompletionExecutor.executeCommand(CompletionExecutor.java:188)
        at com.oracle.graal.pointsto.util.CompletionExecutor.lambda$executeService$0(CompletionExecutor.java:172)
        at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1395)
        at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
        at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
        at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
        at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
        at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
Caused by: java.lang.ClassCastException: class jdk.vm.ci.meta.ValueKind$IllegalKind cannot be cast to class com.oracle.svm.core.graal.llvm.util.LLVMUtils$LLVMKind (jdk.vm.ci.meta.ValueKind$IllegalKind is in module jdk.internal.vm.ci of loader 'bootstrap'; com.oracle.svm.core.graal.llvm.util.LLVMUtils$LLVMKind is in unnamed module of loader 'app')
        at com.oracle.svm.core.graal.llvm.NodeLLVMBuilder.setResult(NodeLLVMBuilder.java:763)
        at jdk.internal.vm.compiler/org.graalvm.compiler.nodes.ConstantNode.generate(ConstantNode.java:152)
        at com.oracle.svm.core.graal.llvm.NodeLLVMBuilder.emitNode(NodeLLVMBuilder.java:327)
        at com.oracle.svm.core.graal.llvm.NodeLLVMBuilder.doBlock(NodeLLVMBuilder.java:287)
        ... 16 more
```

**Steps to reproduce the issue**
Please include both build steps as well as run steps

hello/HelloWorld.java
```
package hello;

public class HelloWorld {

    public static void main(String[] args) {
        System.out.println("Hello World!!");
        System.exit(0);
    }

}
```
Steps:
```
export GRAALVM_HOME=~/Downloads/graalvm-ce-java17-21.3.0-dev/Contents/Home
export JAVA_HOME=$GRAALVM_HOME   

javac hello/HelloWorld.java 

$GRAALVM_HOME/bin/native-image -Djdk.internal.lambda.eagerlyInitialize=false -H:+ExitAfterRelocatableImageWrite --features=org.graalvm.home.HomeFinderFeature -H:-UseCAPCache -H:CompilerBackend=llvm -H:-SpawnIsolates -H:PageSize=16384 -Dsvm.targetName=iOS -Dsvm.targetArch=arm64 -Dsvm.platform=org.graalvm.nativeimage.Platform\$IOS_AARCH64 -cp . hello.HelloWorld 
```

**Describe GraalVM and your environment:**
 - GraalVM version (latest snapshot builds can be found [here](https://github.com/graalvm/graalvm-ce-dev-builds/releases)), or commit id if built from source: **CE 21.3.0-dev-20210910_2147**
  - JDK version: **JDK17**
 - OS and OS Version: **macOS Big Sur**
 - Architecture: **amd64**

I've also tested a few other older dev versions like 21.3.0-dev-20210826_0017.

**More details**
Consider adding the `--native-image-info` and  `--verbose` flags when building your native image and paste output below.

native_library_info_20210912_165551.txt
```
Static libraries:
   $GRAALVM_HOME/lib/svm/clibraries/darwin-amd64/liblibchelper.a
   .$GRAALVM_HOME/lib/svm/clibraries/darwin-amd64/libdarwin.a
   $GRAALVM_HOME/lib/svm/clibraries/darwin-amd64/libjvm.a
Other libraries: m,z,-framework Foundation,pthread,dl
```
target_info_20210912_165539.txt
```
Building image for target platform: org.graalvm.nativeimage.Platform$IOS_AARCH64
Using native toolchain:
   Name: LLVM (clang)
   Vendor: apple
   Version: 12.0.5
   Target architecture: x86_64
   Path: /usr/bin/cc
Using CLibrary: com.oracle.svm.core.c.libc.NoLibC
```

Add any other information about the problem here. Especially important are stack traces or log output. Feel free to link to gists or to screenshots if necessary.

Full output:
```
[hello.helloworld:13759]    classlist:   1,207.69 ms,  0.96 GB
[hello.helloworld:13759]        (cap):   3,161.91 ms,  0.96 GB
[hello.helloworld:13759]        setup:   4,674.10 ms,  0.96 GB
[hello.helloworld:13759]     (clinit):     257.19 ms,  1.77 GB
[hello.helloworld:13759]   (typeflow):   3,452.09 ms,  1.77 GB
[hello.helloworld:13759]    (objects):   7,814.53 ms,  1.77 GB
[hello.helloworld:13759]   (features):     696.41 ms,  1.77 GB
[hello.helloworld:13759]     analysis:  12,752.97 ms,  1.77 GB
[hello.helloworld:13759]     universe:   1,197.52 ms,  2.37 GB
[hello.helloworld:13759]      (parse):     865.31 ms,  2.37 GB
[hello.helloworld:13759]     (inline):   1,211.55 ms,  2.37 GB
[hello.helloworld:13759]    (compile):  13,692.38 ms,  4.39 GB
[hello.helloworld:13759]      compile:  15,899.38 ms,  4.39 GB
Fatal error:org.graalvm.compiler.graph.GraalGraphError: java.lang.ClassCastException: class jdk.vm.ci.meta.ValueKind$IllegalKind cannot be cast to class com.oracle.svm.core.graal.llvm.util.LLVMUtils$LLVMKind (jdk.vm.ci.meta.ValueKind$IllegalKind is in module jdk.internal.vm.ci of loader 'bootstrap'; com.oracle.svm.core.graal.llvm.util.LLVMUtils$LLVMKind is in unnamed module of loader 'app')
        at node: 27|Constant(illegal, ILLEGAL) { isDefaultStable=false, stableDimension=0, rawvalue=illegal, stampKind=ILLEGAL, stamp=ILLEGAL, value=illegal,  }
        at method: byte[] com.sun.crypto.provider.GHASH.digest()  [Direct call from int GaloisCounterMode$GCMEncrypt.doFinal(byte[], int, int, byte[], int)]
        at com.oracle.svm.core.graal.llvm.NodeLLVMBuilder.doBlock(NodeLLVMBuilder.java:292)
        at com.oracle.svm.core.graal.llvm.SubstrateLLVMBackend.generate(SubstrateLLVMBackend.java:148)
        at com.oracle.svm.core.graal.llvm.SubstrateLLVMBackend.emitLLVM(SubstrateLLVMBackend.java:124)
        at com.oracle.svm.core.graal.llvm.SubstrateLLVMBackend.emitBackEnd(SubstrateLLVMBackend.java:104)
        at jdk.internal.vm.compiler/org.graalvm.compiler.core.GraalCompiler.compile(GraalCompiler.java:146)
        at jdk.internal.vm.compiler/org.graalvm.compiler.core.GraalCompiler.compileGraph(GraalCompiler.java:130)
        at com.oracle.svm.hosted.code.CompileQueue.defaultCompileFunction(CompileQueue.java:1210)
        at com.oracle.svm.hosted.code.CompileQueue.doCompile(CompileQueue.java:1172)
        at com.oracle.svm.hosted.code.CompileQueue$CompileTask.run(CompileQueue.java:279)
        at com.oracle.graal.pointsto.util.CompletionExecutor.executeCommand(CompletionExecutor.java:188)
        at com.oracle.graal.pointsto.util.CompletionExecutor.lambda$executeService$0(CompletionExecutor.java:172)
        at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1395)
        at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
        at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
        at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
        at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
        at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
Caused by: java.lang.ClassCastException: class jdk.vm.ci.meta.ValueKind$IllegalKind cannot be cast to class com.oracle.svm.core.graal.llvm.util.LLVMUtils$LLVMKind (jdk.vm.ci.meta.ValueKind$IllegalKind is in module jdk.internal.vm.ci of loader 'bootstrap'; com.oracle.svm.core.graal.llvm.util.LLVMUtils$LLVMKind is in unnamed module of loader 'app')
        at com.oracle.svm.core.graal.llvm.NodeLLVMBuilder.setResult(NodeLLVMBuilder.java:764)
        at jdk.internal.vm.compiler/org.graalvm.compiler.nodes.ConstantNode.generate(ConstantNode.java:152)
        at com.oracle.svm.core.graal.llvm.NodeLLVMBuilder.emitNode(NodeLLVMBuilder.java:328)
        at com.oracle.svm.core.graal.llvm.NodeLLVMBuilder.doBlock(NodeLLVMBuilder.java:288)
        ... 16 more
[hello.helloworld:13759]      [total]:  36,022.06 ms,  4.39 GB
```

**Comment by rodrigar-mx** [comment]
Thanks for sharing @jperedadnr. We will take a closer look to it and get back to you.

**Comment by rodrigar-mx** [comment]
@jperedadnr native-image in graalvm jdk17 is in an experimental status, still I have raised a ticket internally with the dev team and they will answer you accordingly.

**Comment by loicottet** [comment]
Fixed by a7da464c538fc039289954af837c552c15cdbf88
